consensus/istanbul/core/core.go

```diff
@@ -679,7 +684,36 @@ func (c *core) resendRoundChangeMessage() {
 }
 
 func (c *core) checkValidatorSignature(data []byte, sig []byte) (common.Address, error) {
-	return istanbul.CheckValidatorSignature(c.current.ValidatorSet(), data, sig)
+	logger := c.newLogger("func", "checkValidatorSignature")
+
+	var signerAddress common.Address
+	cacheKey := msgSigCacheKey{msgData: data, msgSig: sig}
+	cacheKeyBytes, err := rlp.EncodeToBytes(cacheKey)
```

Comment:
Nit: I'm a little worried about the overhead of this, but I think if there's an improvement in perf it's because `checkValidatorSignature` is used also is in `handleMsg` (via `c.validateFn`)  which checks the signature for all backlog events (& nodes are pretty aggressive about backlogging messages even in the current seq & round).

That to me seems like it occurs much more often in the happy case than round changes (which this should also help with).